Fold WordPressRating into Simplenote

mokagio opened the pull request:
The library has been made read-only _in 2018_. There was no point in adding SwiftPM support to it to fetch it in what is likely its only client. The fact that it had only two classes made folding it a no brainer.

See https://linear.app/a8c/issue/AINFRA-447

> [!NOTE]  
> UI tests failing because of https://github.com/Automattic/simplenote-ios/issues/1717


### Test

Commented the gate for the review-flow view, so that it appears on every launch:

```diff
diff --git a/Simplenote/Classes/SPNoteListViewController.m b/Simplenote/Classes/SPNoteListViewController.m
index d0c0db47..dd93ef9a 100644
--- a/Simplenote/Classes/SPNoteListViewController.m
+++ b/Simplenote/Classes/SPNoteListViewController.m
@@ -651,9 +651,9 @@
 
 - (void)showRatingViewIfNeeded
 {
-    if (![[SPRatingsHelper sharedInstance] shouldPromptForAppReview]) {
-        return;
-    }
+//    if (![[SPRatingsHelper sharedInstance] shouldPromptForAppReview]) {
+//        return;
+//    }
     
     [SPTracker trackRatingsPromptSeen];
```

<img alt="image" src="https://github.com/user-attachments/assets/bf200f94-70c1-48e8-9240-73ba81f769e6" width=300/>

Selected "I like it" and verified the App Store review prompt is loaded.

<img alt="image" src="https://github.com/user-attachments/assets/b25b7940-8a0d-430a-9313-5c9fbb8efcd9" width=300/>


Given the source code for the rating helper library was not changed, this for me was enough of a test to ensure everything is wired together alright.

### Review

Only one developer  required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

mokagio committed:
Copy the two WordPressRatings types instead of using CocoaPods

The library has been made read-only _in 2018_. There was no point in
adding SwiftPM support to it to fetch it in what is likely its only
client. The fact that it had only two classes made folding it a no
brainer.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented on Simplenote.xcodeproj/project.pbxproj:
```diff
@@ -1346,6 +1346,10 @@
+/* Begin PBXFileSystemSynchronizedRootGroup section */
+		3FA8C2862DFBAF7900E1F4A3 /* WordPressRatings */ = {isa = PBXFileSystemSynchronizedRootGroup; explicitFileTypes = {}; explicitFolders = (); path = WordPressRatings; sourceTree = "<group>"; };
+/* End PBXFileSystemSynchronizedRootGroup section */
+
```
Notice I imported the files via a synchronized folder. Everything new that we bring into our Xcode projects should be in synchronized folders IMHO. A more light weight way to track files in the project, with far less changes of merge conflicts when new files are added/renamed/removed.

AliSoftware commented:
[Prototype Build is failing with:](https://buildkite.com/automattic/simplenote-ios/builds/1290/steps/canvas?sid=019766d4-aab1-47e6-a175-a2f97b32ba62#019766d4-aae3-4869-b3fc-a2aea12d06c7/309-631)

```
ArgumentError - [Xcodeproj] Unable to find compatibility version string for object version `70`.
```
I wonder if it's because of the use of Synchronized Folders vs the version of Xcode used in CI (`xcode-16.1-v4`)?

AliSoftware commented on Simplenote.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 70;
```
This requires a more recent Xcode version than the `16.1` one currently used by CI, so we might want to upgrade the `IMAGE_ID` too

mokagio committed:
Use Xcode 16.4 in CI

mokagio commented:
Very embarrassed of not having noticed the CI failures... I was focusing on the XCFramework upgrade and just _assumed_ the red was the usual UI tests... 🤦‍♂️

mokagio committed:
Set Xcode project compatibility version to "Xcode 16.0" (77)

This works around an issue in CocoaPods, where `xcodeproj` can't work
with version 70. See https://github.com/CocoaPods/CocoaPods/issues/12671

Version 70 was set when adding the first synchronized folder in
0efd180200fae871b8841006abd37c4

mokagio commented:
CI is now green, other than for the expected UI tests. Thanks @AliSoftware
